### ISSUE TITLE
Pre-commit setup of nodejs results in error as nodejs cannot find GLIBC on Ubuntu 18.04

### ISSUE THREAD

**Issue by paulbrodersen** [networkx/networkx#6754]
### Current Behavior

Running the pre-commit hook 

```shell
pre-commit run --all-files
```

results in an error. Full error trace included in the [pre-commit.log](https://github.com/networkx/networkx/files/11833025/pre-commit.log).
The issue arises possibly to [due the version of Node.js being used](https://stackoverflow.com/questions/72921215/getting-glibc-2-28-not-found). 

### Expected Behavior

Running the pre-commit should not result in an error.

### Steps to Reproduce

Follow the steps outlined in CONTRIBUTING.rst exactly. 

### Environment

Python version: 3.10
NetworkX version: 3.2rc0.dev0

**Comment by jarrodmillman** [comment]
I am not seeing this error on my computer or on our CI system.

What version of pre-commit are you using? Did you try the steps listed in the StackOverflow article?

**Comment by paulbrodersen** [comment]
pre-commit 3.3.3, and no, as I assumed node was being downloaded in the temporary virtualenv being created. I don't use node so I haven't ever installed it (knowingly).

**Comment by MridulS** [comment]
This is probably coming from our `prettier` pre-commit hook and nothing to do with black. There is indeed an issue that could have happen with Windows based systems https://github.com/pre-commit/mirrors-prettier/issues/18#issuecomment-1135071907 but I can't reproduce/debug this for a linux based one.

Could you please try manually removing prettier pre-commit hook from the .pre-commit-config.yml file in the root directory of networkx? After removing it should look something like this.


```
repos:
  - repo: https://github.com/psf/black
    rev: 23.3.0
    hooks:
      - id: black
  - repo: https://github.com/adamchainz/blacken-docs
    rev: 1.13.0
    hooks:
      - id: blacken-docs
  - repo: https://github.com/charliermarsh/ruff-pre-commit
    rev: v0.0.258
    hooks:
      - id: ruff
        args:
          - --fix
 ```

Does this bit run for you?

**Comment by MridulS** [comment]
Also how recent is your OS distribution? Node 18+ prebuilt binaries need a new version of glibc.

**Comment by paulbrodersen** [comment]
@MridulS 

> Does this bit run for you?

It does. No errors or (undue) warnings when making commits with the new config. 

> Also how recent is your OS distribution? Node 18+ prebuilt binaries need a new version of glibc.

You have a point there as well. First (well, second) thing I tried was updating glibc. However, I (just now) checked and it  turns out that the newest available version 2.27.

**Comment by MridulS** [comment]
> You have a point there as well. First (well, second) thing I tried was updating glibc. However, I (just now) checked and it turns out that the newest available version for Ubuntu 18.04 is version 2.27.

Yeah unfortunately the latest node versions depend on version 2.28+ https://github.com/nodejs/node/blob/main/BUILDING.md#platform-list, there is indeed a way of overriding the node version in pre-commit  https://pre-commit.com/#overriding-language-version but we shouldn't be doing this in the main networkx repo. 

Also ubuntu 18.04 is pretty much EOL https://endoflife.date/ubuntu so I would recommend updating anyway :)

I'll close this issue, thanks for reporting!

**Comment by paulbrodersen** [comment]
> Also ubuntu 18.04 is pretty much EOL https://endoflife.date/ubuntu so I would recommend updating anyway :)

I am painfully aware as I have already updated our servers but I haven't got around to it on my personal laptop as my toddler is convinced that he deserves my full attention whenever I am home (and he does. ;-)).
